### ISSUE TITLE
CA1508 (Avoid dead conditional code) - false positive in nested foreach

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn-analyzers#5006]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

False positive in following code:
```c#
static void M(List<C> list1, List<C> list2)
{
    foreach (var c1 in list1)
    {
        if (c1.X == "")
        {
            foreach (var c2 in list2)
            {
                if (c2.X == "") //CA1508: 'c2.X == ""' is always 'true'
                { }
            }
        }
    }
}
class C { public string X; }
```

### Expected behavior
Diagnostic is not reported in this case.

**Comment by mavasani** [comment]
This does not repro on latest .NET 8 preview SDK
